### PR TITLE
Upon `blitz new`, prompt user for upgrade if blitz version is outdated

### PR DESCRIPTION
**Pull request by abuuzayr**
Closes: #1017 

### What are the changes and their implications?

- If blitz global package is outdated, it will ask the user if they wish to upgrade

image

*Note:* While this is working and ready for review, I have some questions:

- Do we need to check if the global package was installed by `yarn` or `npm`? i.e. if it was installed by `npm` and `yarn` is installed, upgrade via `npm` instead of `yarn`. 
- Is testing the output required and possible? I saw in `new.test.ts` that the tests do not cover everything (e.g. that all the flags are working as intended) and I had some difficulty on writing tests against the code that I am adding, so I am wondering if this is something that is possible and that the tests should be more comprehensive.

### Checklist

- [ ] Tests added for changes
- [ ] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes

### CODE REVIEW

**Comment by flybayer** [comment]
Thanks!

> * Do we need to check if the global package was installed by `yarn` or `npm`? i.e. if it was installed by `npm` and `yarn` is installed, upgrade via `npm` instead of `yarn`.

Yes, this is important because if installed globally by both, the behavior is indeterminate.

> * Is testing the output required and possible? I saw in `new.test.ts` that the tests do not cover everything (e.g. that all the flags are working as intended) and I had some difficulty on writing tests against the code that I am adding, so I am wondering if this is something that is possible and that the tests should be more comprehensive.

Yeah not sure exactly how to write a test for this either. You can skip that, will just have to test manually.

**Comment by abuuzayr** [comment]
Added check for whether the global package is installed by `yarn` or `npm` first before running upgrade. I opted to check `yarn` first because `npm list -g` takes 22s to return a response, with or without `--depth=0 --silent`, at least on my machine. `yarn global list` takes 2s. 

If we should check npm first, do let me know!

File: packages/cli/src/commands/new.ts [hunk 1]
```diff
@@ -50,13 +53,74 @@ export class New extends Command {
...
+          {
+            name: "no",
+            message: `No - Continue with old version (${this.config.version}) - NOT recommended`,
+          },
+        ]
+
+        const promptUpgrade: any = await this.enquirer.prompt({
+          type: "select",
+          name: "upgrade",
+          message: "Your blitz CLI version is outdated. Upgrade?",
```
Comment (flybayer) [inline packages/cli/src/commands/new.ts:81]:
```suggestion
          message: "Your global blitz version is outdated. Upgrade?",
```

**Comment by flybayer** [comment]
Sweet, looks good. Thank you!!

**Pull request merged**